src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -516,12 +517,12 @@ private AuthenticationResult postMessage(String requestMessage, HashMap<String,
 
                 Logger.v(TAG, "Server error message:" + errMessage);
                 if (response.getResponseException() != null) {
-                    throw response.getResponseException();
+                    throw new IOException(response.getResponseException());
```

Comment:
Should be AuthenticationServerProtocolException.  There is no reason for the app to crash in this case.